iree/test/e2e/xla/reverse.mlir

```diff
@@ -1,14 +1,16 @@
-// RUN: iree-run-mlir -iree-hal-target-backends=vmla %s | IreeFileCheck %s
-// RUN: [[ $IREE_VULKAN_DISABLE == 1 ]] || (iree-run-mlir -iree-hal-target-backends=vulkan-spirv %s | IreeFileCheck %s)
+func @xla_reverse() attributes { iree.module.export } {
+  %t1 = iree.unfoldable_constant dense<[[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]]> : tensor<2x3xf32>
 
-// CHECK-LABEL: EXEC @xla_reverse
-func @xla_reverse () -> (tensor<2x3xf32>, tensor <2x3xf32>, tensor <2x3xf32>) {
-  %t1 = iree.unfoldable_constant dense<[[1.0e0, 2.0e0, 3.0e0], [4.0e0, 5.0e0, 6.0e0]]> : tensor<2x3xf32>
-  %0 = "xla_hlo.reverse"(%t1) {dimensions = dense<[0]> : tensor<1xi64>} : (tensor<2x3xf32>) -> tensor<2x3xf32>
-  %1 = "xla_hlo.reverse"(%t1) {dimensions = dense<[1]> : tensor<1xi64>} : (tensor<2x3xf32>) -> tensor<2x3xf32>
-  %2 = "xla_hlo.reverse"(%t1) {dimensions = dense<[0, 1]> : tensor<2xi64>} : (tensor<2x3xf32>) -> tensor<2x3xf32>
-  return %0, %1, %2: tensor<2x3xf32>, tensor<2x3xf32>, tensor<2x3xf32>
+  %dim0 = "xla_hlo.reverse"(%t1) {dimensions = dense<[0]> : tensor<1xi64>} : (tensor<2x3xf32>) -> tensor<2x3xf32>
+  %expected_dim0 = iree.unfoldable_constant dense<[[4.0, 5.0, 6.0], [1.0, 2.0, 3.0]]> : tensor<2x3xf32>
+  check.expect_eq(%dim0, %expected_dim0) : tensor<2x3xf32>
```

Comment:
Since it's not actually doing math, I thought `expect_eq` made sense instead of `expect_almost_eq`. Reverse shouldn't change the values, right? Or do we not want to promise that?

Reply:
Pretty much any float should be using expect almost eq unless it's a low level test. It's really hard to ensure that bits don't change in the general case when coming from this level: if the float was quantized to a fixed point number because the target platform doesn't support floats it's not likely to match up. It's a good habit to just make them all almost eq.

Reply:
Done.